vllm/v1/worker/gpu_model_runner.py

```diff
@@ -2364,10 +2364,25 @@ def capture_model(self) -> None:
         start_time = time.perf_counter()
         start_free_gpu_memory = torch.cuda.mem_get_info()[0]
 
+        @contextmanager
+        def freeze_gc():
+            # Optimize garbage collection during CUDA graph capture.
+            # Clean up, then freeze all remaining objects from being included
+            # in future collections.
+            gc.collect()
+            should_freeze = not envs.VLLM_ENABLE_CUDAGRAPH_GC
+            if should_freeze:
+                gc.freeze()
+            try:
+                yield
+            finally:
+                if should_freeze:
+                    gc.unfreeze()
```

Comment:
Actually, I think we should have this in `utils.py` or something like that. The model runner is becoming bloated.

Reply:
Okay I'll try to do a separate PR that will consolidate with the current implementation in the piecewise backend